src/main/java/calculator/util/Parser.java

```diff
@@ -0,0 +1,74 @@
+package calculator.util;
+
+import calculator.constants.DelimiterConstants;
+import calculator.domain.UserInput;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Parser {
+
+    public static boolean hasCustomDelimiter(String string){
+        if(string == null || string.length() < DelimiterConstants.CUSTOM_DELIMITER_BOUND_LENGTH+1) {
+            return false;
+        }
+
+        if(!string.startsWith(DelimiterConstants.CUSTOM_DELIMITER_PREFIX)) {
+            return false;
+        }
+
+        int endIndex = string.indexOf(DelimiterConstants.CUSTOM_DELIMITER_SUFFIX, 2);
+        return endIndex > 2;
+    }
+
+    public static String extractCustomDelimiter(String string){
+        StringBuilder delimiter = new StringBuilder();
+
+        for(int index = 2; index < string.length()-1; index++){
+            if(string.startsWith(DelimiterConstants.CUSTOM_DELIMITER_SUFFIX, index)) {
+                break;
+            }
+
+            delimiter.append(string.charAt(index));
+        }
+
+        return delimiter.toString();
+    }
+
+    public static List<Integer> parseNumbersFromInput(UserInput userInput) {
+        List<Integer> result = new ArrayList<>();
+        StringBuilder number = new StringBuilder();
+
+        String string = userInput.input();
+        List<String> delimiters = userInput.delimiters();
+
+        int delimiterLength = userInput.hasCustomDelimiter() ? delimiters.getFirst().length() : 1;
```

Comment:
```suggestion
        int delimiterLength = delimiters.getFirst().length();
```
이 부분으로 대체할 수 있지 않을까요? 불필요한 연산 같아요!

Reply:
오 이제보니 단순화할 수 있던 부분이네요! 깔끔한 코드 피드백 감사합니당